### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by Nesze**

### CODE REVIEW

File: model.go [hunk 1]
```diff
@@ -16,7 +16,7 @@ const (
 	EventTypeProfileReset         EventType = "$profile_reset"
 	EventTypePasswordResetRequest EventType = "$password_reset_request"
 	EventTypeChallenge            EventType = "$challenge"
-	EventTypeLogout               EventType = "&logout"
```
Comment (pansbro12) [inline model.go:19]: good spot

File: README.md [hunk 2]
```diff
@@ -15,3 +15,30 @@ go get github.com/utilitywarehouse/castle-go
...
+The pkg wraps the two [Risk Assessment endpoints](https://reference.castle.io/#tag/risk_assessment) of the Castle API: Risk and Filter.
+
+The difference between the two are better explained in the [docs](https://docs.castle.io/docs/integration-guide):
+
+> The biggest difference between the Risk API and the Filter API is the former is used for checking a user that has successfully logged in (so you measure risk of their actions in your app), whereas the latter is used to check visitors, before they log in (so you filter out abusive behavior).
+
+The right usage of the Filter API and rest of the technical differences are laid out [here](https://docs.castle.io/docs/anonymous-activity).
+
+All in all, use Filter API for anonymous user events, and Risk API for logged in users.
+The model of the endpoints is almost the same, the request is almost identical while the response is 100% is. Both return risk assessment scores, which depending on the flow (event and status) might be ignored.
```
Comment (Jarota) [inline README.md:30]:
```suggestion
The model of the endpoints is almost the same, the request is almost identical while the response is 100%. Both return risk assessment scores, which depending on the flow (event and status) might be ignored.
```

File: README.md [hunk 3]
```diff
@@ -15,3 +15,30 @@ go get github.com/utilitywarehouse/castle-go
...
+
+> Scores are computed in real time from the data sent via the Risk and Filter APIs
+[1](https://docs.castle.io/docs/risk-scoring)
+
+> Note that these can also be sent to the Log API, but that would degrade risk scoring performance since the risk score isn't evaluated for Log events
+[2](https://docs.castle.io/docs/anonymous-activity)
+
+## Repo
+
+Originally forked from [castle/castle-go](https://github.com/castle/castle-go) now it lives on its own. The original repo has not been maintained, and as of today only supports long deprecated Castle APIs.
```
Comment (Jarota) [inline README.md:44]: This is a bit strange, that it's not maintained at all? 🤔 It's listed [here](https://docs.castle.io/docs/sdk-download) as an 'officially' supported SDK but looks like even that docs page hasn't been updated in ~2 years ...

File: client.go [hunk 4]
```diff
@@ -154,23 +93,26 @@ func (c *Castle) sendRiskCall(ctx context.Context, e *castleAPIRequest) (Recomme
...
+	switch action {
+	case "allow":
+		return RecommendedActionAllow
+	case "deny":
+		return RecommendedActionDeny
+	case "challenge":
+		return RecommendedActionChallenge
+	default:
+		return RecommendedActionNone
 	}
```
Comment (Jarota) [inline client.go:117]: Just double checking - do we no longer need/want to return errors in these cases, and `RecommendedActionNone` is sufficient?

Comment (Nesze) [reply client.go:117]: These two checks are redundant. We check for error above by looking at the http.StatusCode. They seem to me to be some left-overs.

File: README.md (hunk 3 above)
Comment (Nesze) [reply README.md:44]: It's worth clarifying with them, but last commit was 4 years ago. While no support for any of their current APIs, i.e. Risk, Filter and Log, only older deprecated ones.

https://github.com/castle/castle-go

**Review by pansbro12 — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (Jarota) [reply README.md:44]: Interesting yeah maybe we could reach out or even collaborate on it. Do you think Ryan is a good first port of call?

**Review by Jarota — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (Nesze) [reply README.md:44]: Yeah. A few other things also came up in these past few days that we'd like to clarify, so guess just need to collect them.

**Pull request merged**